### PR TITLE
Add function get_last_section_by_instructor_and_terms

### PR DESCRIPTION
**Pull request by fanglinfang**

### CODE REVIEW

**Comment by jlaney** [comment]
What is the purpose of this function?

**Comment by fanglinfang** [comment]
The purpose  is to know if someone has taught in a number of terms (-2, -1, current, +1, +2) and the last term she has taught.

File: uw_sws/section.py [hunk 1]
```diff
@@ -190,8 +190,57 @@ def _get_sections_by_person_and_term(person,
...
+    section = SectionReference(
+        term=section_term,
+        curriculum_abbr=section_data["CurriculumAbbreviation"],
+        course_number=section_data["CourseNumber"],
+        section_id=section_data["SectionID"],
+        url=section_data["Href"])
+    return section
+
+
+def _get_sections_by_person_and_term(person,
```
Comment (jlaney) [inline uw_sws/section.py]: Why are you keeping _get_sections_by_person_and_term around? It doesn't seem to be adding anything useful.

File: uw_sws/section.py [hunk 2]
```diff
@@ -190,8 +192,34 @@ def _get_sections_by_person_and_term(person,
         params.append(("delete_flag", ','.join(sorted(delete_flag)),))
 
     url = "%s?%s" % (section_res_url_prefix, urlencode(params))
+    return get_resource(url)
```
Comment (ezturner) [inline uw_sws/section.py]: Why did we remove the sectionref conversion here?

File: uw_sws/section.py [hunk 3]
```diff
@@ -190,8 +192,34 @@ def _get_sections_by_person_and_term(person,
...
+                                                    term,
+                                                    "Instructor",
+                                                    False,
+                                                    future_terms,
+                                                    transcriptable_course,
+                                                    delete_flag)
+    except DataFailureException as ex:
+        if ex.status == 404:
+            return None
+        raise
```
Comment (ezturner) [inline uw_sws/section.py]: Should this be:
raise ex

rather than:
raise
?

File: uw_sws/section.py (hunk 2 above)
Comment (jlaney) [reply uw_sws/section.py]: The new function get_last_section_by_instructor_and_terms wants to modify the returned json before models are built, as an optimization.

File: uw_sws/section.py (hunk 3 above)
Comment (jlaney) [reply uw_sws/section.py]: That is what it's doing, calling raise with no args raises the last exception.

**Comment by fanglinfang** [comment]
Thanks much!

**Pull request merged**